assets/demo_maps/office.building.yaml

```diff
@@ -181,7 +181,7 @@ levels:
       - [814.828, 239.25, 0, "", {is_parking_spot: [4, false]}]
       - [1210.544, 365.254, 0, ""]
       - [1191.442, 824.457, 0, ""]
-      - [1232.421, 658.567, 0, tinyRobot1_charger, {is_charger: [4, true], is_holding_point: [4, true], is_parking_spot: [4, true], spawn_robot_name: [1, tinyRobot1], spawn_robot_type: [1, TinyRobot]}]
+      - [1232.421, 658.567, 0, tinyRobot1_charger, {is_charger: [4, true], is_holding_point: [4, true], is_parking_spot: [4, true], spawn_robot_name: [1, tinyRobot1], spawn_robot_type: [1, Luca/TinyRobot]}]
```

Comment:
I'm not 100% convinced by this change.
At least for me, the idea of this demo map is that it is as close as possible to a legacy traffic editor world and shows how we can easily import models from the legacy format. Doing this change here will make this specific map work but every other traffic editor map still won't have any robots.
What I would propose instead is to add the fuel organization when the legacy map is converted to a site. Now the question will be "which organization"? The answer is "it's complicated" but for now I would just add an `OpenRobotics/` in front of every loaded model. There does seem to be an [OpenRobotics/TinyRobot](https://app.gazebosim.org/OpenRobotics/fuel/models/TinyRobot) model on fuel anyway. In the future if we can't upload models to `OpenRobotics` we can just change to a different organization

Reply:
I was meaning to ask, how did we end up with a situation where [decorative models use `OpenRobotics/`](https://github.com/open-rmf/rmf_demos/blob/9bdd28ff2d326634c27536a2cd120290f15c4356/rmf_demos_maps/maps/office/office.building.yaml#L113) but [robots don't](https://github.com/open-rmf/rmf_demos/blob/9bdd28ff2d326634c27536a2cd120290f15c4356/rmf_demos_maps/maps/office/office.building.yaml#L113)?

I assume it has something to do with pit crew and/or fuel versus [`rmf_demos_assets`](https://github.com/open-rmf/rmf_demos/tree/main/rmf_demos_assets/models)? I'm going to guess the rationale is that decorative models are stable enough that fetching them from fuel will generally be fine, whereas robot models may need to have their plugin parameters tweaked, e.g. changing speeds or bidirectionality for the slotcar plugin?

If that's the case then I'd suggest we add the following as an action item for #254 :
1. Upload all the models from `rmf_demos_assets` to the `OpenRobotics` as @luca-della-vedova suggested
2. When loading a model, check:
    - Does the loaded SDF model contain a slotcar plugin?
      - If it does have a slotcar plugin, check if the entity already has a `Mobility` component
        - If it does already have a `Mobility` component, do nothing
        - If it doesn't already have a `Mobility` component, check if it has an `IsStatic` component
          - If it does have an `IsStatic` component, do nothing
          - If it doesn't have an `IsStatic` component, add a `Mobility` component with a `DifferentialDrive` configuration
      - If it does not have a slotcar plugin, do nothing

In the meantime for this PR I think it's enough to just change this to `OpenRobotics/TinyRobot` for the demo site.

Reply:
> In the meantime for this PR I think it's enough to just change this to `OpenRobotics/TinyRobot` for the demo site.

Fully agree with the analysis (including possible historical causes) and outlined solution except this teeny tiny point!
My proposal was to leave it as is on the `.building.yaml` level and add a conversion step that prepends `OpenRobotics/` to the value under `spawn_robot_type` so it would import a legacy `spawn_robot_type: [1, TinyRobot]` into a `AssetSource::Search(OpenRobotics/TinyRobot)`.
The main rationale is that this import strategy would make it possible to automatically import every legacy project out there. By contrast, if we make it necessary to prepend the organization name to the asset name (i.e. what this diff is doing) legacy projects will fail out of the box to fetch models and will have to be updated.

It's basically a tradeoff of:
* Updating our pipeline(s) so they work with existing maps (my proposal) or
* Force users to update existing maps so they work with our pipelines (this diff)

Reply:
> My proposal was to leave it as is on the .building.yaml level and add a conversion step that prepends OpenRobotics/ to the value under spawn_robot_type

Ah yeah, that should give the smoothest outcome for the use cases that we're anticipating. I'd be a little concerned that it could be disruptive for anyone using their own custom robot models.

I don't love what I'm about to suggest, but what if we hard-code the names of the models that we know are hosted at the `OpenRobotics` org and only prepend `OpenRobotics/` to those?

Reply:
Or a more complex but even more backwards compatible alternative:

Introduce a `spawn-robot` scheme that we apply during legacy imports which behaves similarly to `search` except that when the local search fails it will then add `OpenRobotics/` and search fuel.

Reply:
We could also add an additional searching iteration for `Search` assets, this is the current sequence:
https://github.com/open-rmf/rmf_site/blob/fe86373b94621b22c482bebd0b6333da2e8ff496/rmf_site_editor/src/site_asset_io.rs#L228-L232

We could extend the search relative to `MODEL_ENVIRONMENT_VARIABLE` (which is also what is used by Gazebo to look for models) to try the path without the organization, if an organization is present.
This means that if an `OpenRobotics/Table` is requested, the `Search` will look for both a `[GZ_SIM_PATH]/OpenRobotics/Table` as well as `[GZ_SIM_PATH]/Table`.
This could be undesirable if users have both but I would argue that at that point it is unclear to me how Gazebo itself would behave, since it would mean there are duplicated models in its path.
This search would still have priority and happen before the Fuel search and hopefully be fast enough (it's only a local file lookup) that it won't generally be noticeable.

Reply:
A part of me is scared of adding more complexity to `search` because it could reach a point where the rules become very unclear to users and it becomes a cargo cult of people trying different weird things until they can get the search to work. But I'm open to that approach for the sake of helping existing users transition.

Reply:
After some more thought I agree that all these solutions are a bit messy and probably unifying the robot type parameters going forward is the best solution.
I uploaded all the demo robots to a new [Open-RMF organization](https://app.gazebosim.org/Open-RMF), as well as opened an upstream [PR](https://github.com/open-rmf/rmf_demos/pull/272) in `rmf_demos` to change the demo maps going forward to use the same convention as models.
This will mean that pre-existing third party maps might have import issues but at least `rmf_demos` and `rmf_site` will be in a consistent state.
For this specific PR, as long as the `Luca/` changes to `Open-RMF/` it should all be good

Reply:
I'm a little bit hesitant to remove `rmf_demos_assets` entirely...

1. Where are we actually version controlling those models? Will the copies on fuel be the only source? Is that the workflow recommended by the Gazebo team?
2. traffic-editor doesn't currently have a reasonable way to override the slotcar plugin properties, so traffic-editor users will be locked into the current slotcar parameters for the models on fuel

If it's possible for the `rmf_demos_assets` models to be installed locally in a way that they could be found via both searches for `Open-RMF/TinyRobot` and searches for `TinyRobot` then I think that would offer a cleaner transition.

This would be a good topic to discuss at the PMC session tomorrow morning.

Reply:
Yea I can see the controversy, let's discuss the options in the PMC. For this PR though @xiyuoh can you just remove `Luca` and change to `Open-RMF`? We should do that regardless of what we end up doing with `rmf_demos_assets`

Reply:
[5ce02a1](https://github.com/open-rmf/rmf_site/pull/242/commits/5ce02a131604dd8c4fe1eb0f5448310f85360d13)